BookStoreAPI.Tests/IntegrationTests/Books/BooksTests.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System.Net;
+using System.Text;
+using AutoMapper;
+using BookStoreAPI.Functions.Commands.Book.Create;
+using BookStoreAPI.Functions.Commands.Book.Patch;
+using BookStoreAPI.Models;
+using BookStoreAPI.Tests.Utils;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace BookStoreAPI.Tests.IntegrationTests.Books;
+
+public class BooksTests : IClassFixture<BookStoreWebApplicationFactory<Program>>
+{
+    private readonly BookStoreWebApplicationFactory<Program> _factory;
+    private readonly IMapper _mapper;
+    private readonly IServiceProvider _serviceProvider;
+
+    public BooksTests(BookStoreWebApplicationFactory<Program> factory)
+    {
+        _factory = factory;
+
+        _serviceProvider = _factory.Services;
+        _mapper = _serviceProvider.GetRequiredService<IMapper>();
+
+        Data.SeedData(_serviceProvider);
+    }
+
+    [Fact]
+    public async Task Get_ReturnsListOfAllBooksFromDatabase()
+    {
+        using var scope = _factory.Services.CreateScope();
+        var dbContext = scope.ServiceProvider.GetRequiredService<BookStoreDatabaseContext>();
+
+        var client = _factory.CreateClient();
+
+        var response = await client.GetAsync("/api/BookStore");
+        response.EnsureSuccessStatusCode();
+        var contentString = await response.Content.ReadAsStringAsync();
+        var books = JsonConvert.DeserializeObject<List<BookDto>>(contentString);
+
+        var currentBooks = await dbContext.Books.Include(b => b.RentalHistory).Include(b => b.Genre).ToListAsync();
+        var currentBooksDto = _mapper.Map<List<BookDto>>(currentBooks);
+
+        Assert.NotNull(books);
+        Assert.NotEmpty(books);
+        Assert.Equivalent(currentBooksDto, books);
+    }
+
+    [Fact]
+    public async Task Get_ReturnsBookFromDatabase()
+    {
+        using var scope = _factory.Services.CreateScope();
```

Comment:
The AAA (Arrange, Act, Assert) pattern is a common way of writing unit tests for a method under test.

The Arrange section of a unit test method initializes objects and sets the value of the data that is passed to the method under test.

The Act section invokes the method under test with the arranged parameters.

The Assert section verifies that the action of the method under test behaves as expected. For .NET, methods in the [Assert](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert) class are often used for verification.

Reply:
Czyli w skrócie można podzielić na 3 sekcje, które są rozdzielane pustą linią